### PR TITLE
UHM-5050: (Support restricting Lab Results display to certain concepts)

### PR DESCRIPTION
**Pull request by mogoodrich**
The organization of this OWA leaves a lot to be desired, so I struggled a bit where to add this code and what to name it...

### CODE REVIEW

File: app/js/sagas/conceptSaga.js [hunk 1]
```diff
@@ -76,3 +78,41 @@ export function* fetchAndSetConcept(action) {
...
+    // filter out all that aren't lab tests or tests
+    concepts = concepts.filter(c => c.conceptClass.name === 'Test' || c.conceptClass.name === 'LabSet');
+
+    // create a set for easy lookup
+    const conceptSet = new Set();
+    concepts.forEach(c => conceptSet.add(c.uuid));
+
+    yield put(setLabResultsToDisplayConceptSet(conceptSet));
+  } catch (e) {
+    // TODO actually do something in failure case
```
Comment (mogoodrich) [inline app/js/sagas/conceptSaga.js:111]: I confirmed that if the GP doesn't represent a valid concept, everything still works fine.

File: app/js/sagas/conceptSaga.js [hunk 2]
```diff
@@ -76,3 +78,41 @@ export function* fetchAndSetConcept(action) {
...
+    const response = yield call(conceptRest.getConcept, conceptUuid, CONCEPT_SET_REP);
+
+    let concepts = response.setMembers;
+
+    // flatten the concept set
+    while (concepts.some(c => c.set)) {
+      concepts = concepts.flatMap(c => (c.set ? [...c.setMembers, {
+        uuid: c.uuid,
+        set: false,
+        conceptClass: c.conceptClass,
```
Comment (brandones) [inline app/js/sagas/conceptSaga.js:98]:
```suggestion
        ...c,
        set: false,
```

?

**Review by brandones — APPROVED** [review]
Looks good!

**Comment by mogoodrich** [comment]
Thanks guys!  Will take a closer look at your recommendation on Monday @brandones  and likey merge in then.

File: app/js/sagas/conceptSaga.js (hunk 2 above)
Comment (mogoodrich) [reply app/js/sagas/conceptSaga.js:98]: Made this change manually, thanks...

**Pull request merged**